### ISSUE TITLE
Aggregating modules fails when exporting default from another module along with other named exports

### ISSUE THREAD

**Issue by ravijayaramappa** [rollup/rollup#2992]
- Rollup Version: 1.7.4
- Operating System (or Browser): macOS Mojave
- Node Version: 10.16.0

### How Do We Reproduce?
When an aggregating module exports the default from another module along with named properties from a different module, the generated output will through a ReferenceError upon invocation.

Example:
```
export { default } from 'x/foo';
export { namedProp } from 'x/bar';
```
[REPL repro](https://rollupjs.org/repl/?version=1.16.7&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMCU3QiUyMGRlZmF1bHQlMjAlN0QlMjBmcm9tJTIwJ3glMkZmb28nJTNCJTVDbmV4cG9ydCUyMCU3QiUyMG5hbWVkUHJvcCUyMCU3RCUyMGZyb20lMjAneCUyRmJhciclM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJhbWQlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==)

Importing a default and named export from the same module seems to work fine ✅ 
Example:
```
export { default, namedProp } from 'x/foo';
```

### Expected Behavior
amd format
```
define(['exports', 'x/foo', 'x/bar'], function (exports, foo, bar) { 'use strict';
	foo = foo && foo.hasOwnProperty('default') ? foo['default'] : foo;

	exports.default = foo;
	Object.defineProperty(exports, 'namedProp', {
		enumerable: true,
		get: function () {
			return bar.namedProp;
		}
	});
	Object.defineProperty(exports, '__esModule', { value: true });
});

```

### Actual Behavior
amd format
```
define(['exports', 'x/foo', 'x/bar'], function (exports, foo, bar) { 'use strict';
	foo = foo && foo.hasOwnProperty('default') ? foo['default'] : foo;

	exports.default = foo__default; // <-- this will result in a ReferenceError
	Object.defineProperty(exports, 'namedProp', {
		enumerable: true,
		get: function () {
			return bar.namedProp;
		}
	});
	Object.defineProperty(exports, '__esModule', { value: true });
});
```

This could be related to a similar issue reported prior: https://github.com/rollup/rollup/issues/2977

**Comment by lukastaegert** [comment]
It is certainly related to #2977 but it seems to manifest slightly differently so I will keep it as a separate issue. Hope we can fix them together soon!

**Comment by lukastaegert** [comment]
I’m working on a fix for this but due to other issues it may take another week to complete.

**Comment by lukastaegert** [comment]
Fix at #3084
